main-command/src/main/scala/sbt/internal/client/NetworkClient.scala

```diff
@@ -347,9 +347,14 @@ class NetworkClient(
             .directory(arguments.baseDirectory)
             .redirectInput(Redirect.PIPE)
         processBuilder.environment.put(Terminal.TERMINAL_PROPS, props)
-        val process = processBuilder.start()
-        sbtProcess.set(process)
-        Some(process)
+        Try(processBuilder.start()) match {
+          case Success(process) =>
+            sbtProcess.set(process)
+            Some(process)
+          case Failure(e) =>
+            if (log) console.appendLog(Level.Error, s"Failed to start server : $e")
+            None
```

Comment:
Here `None` means the server is already booting up in a different process. So it will wait indefinitely for the socket file to appear.
```suggestion
            throw ServerFailedException
```

Reply:
Thanks for your feedback, your suggestion is now pushed !